tests/integration_tests/security/test_spectre_meltdown.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# N.B.: Although this repository is released under the Apache-2.0, part of its test requires a
+# script from the third party "Spectre & Meltdown Checker" project. This script is under the
+# GPL-3.0-only license.
+"""Tests spectre / meltdown mitigation."""
+
+import pytest
+import requests
+
+from framework import utils
+from host_tools.network import SSHConnection
+
+
+CHECKER_URL = "https://meltdown.ovh"
+CHECKER_FILENAME = "spectre-meltdown-checker.sh"
+
+
+@pytest.fixture(name="spectre_meltdown_checker")
+def download_spectre_meltdown_checker(tmp_path):
+    """Download spectre / meltdown checker script."""
+    resp = requests.get(CHECKER_URL, timeout=5)
+    assert (
+        resp.ok
+    ), f"Failed to download the checker script. Status code: {resp.status_code}"
+
+    path = tmp_path / CHECKER_FILENAME
+    with open(path, "wb") as file:
+        file.write(resp.content)
+
+    return path
+
+
+def test_spectre_meltdown_checker_on_host(spectre_meltdown_checker):
+    """
+    Test with the spectre / meltdown checker on host.
+
+    @type: security
+    """
+    utils.run_cmd(f"sh {spectre_meltdown_checker} --explain")
+
+
+def test_spectre_meltdown_checker_on_guest(
+    spectre_meltdown_checker,
+    test_microvm_with_spectre_meltdown,
```

Comment:
Is this a new image in the artifacts bucket?

Reply:
Yes, it is. To make this check passed, we need to enable kernel config inside guest and install `strings` tool. So I built a new kernel and rootfs and put them under `spec.ccfc.min/<arch>/ubuntu_with_spectre_meltdown`. Should I also commit the change the kernel config (`resources/microvm-kernel-<arch>-<version>.config`) and rootfs (`resources/tests/setup_rootfs.sh`)?

Reply:
Yeah, it would be good to keep track of that.  Is there any downside of enabling these options? I.e. in the future we may be able to consolidate these kernels/rootfs into one.

Reply:
My concern is that the doc tells that the kernerl config under `resources/guest_configs/` are our recommended ones.
https://github.com/firecracker-microvm/firecracker/blob/cd29c64b35f694a5a442a7778b7c1463551bc1e2/docs/rootfs-and-kernel-setup.md?plain=1#L28
I'm not sure if enabling `/proc/config.gz` is recommended for customers.

Reply:
True. From https://wiki.gentoo.org/wiki/Kernel/IKCONFIG_support#Why_is_it_important.3F

> Some people advise against enabling IKCONFIG support as it contains specific information about what features are (or are not) built in to the running kernel, and can pose a security risk if someone with malicious intent gains access to it.

Can we then add a new kernel config `microvm-kernel-ci-arm64-4.14.config` or debug or some such.

Reply:
> Can we then add a new kernel config microvm-kernel-ci-arm64-4.14.config or debug or some such.

Yes, that is a good option! Will fix.

Reply:
Done.